tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -111,11 +111,13 @@ private protected Tracer(TracerManager tracerManager)
         }
 
         /// <summary>
-        /// Gets or sets the global <see cref="Tracer"/> instance.
+        /// Gets the global <see cref="Tracer"/> instance.
         /// Used by all automatic instrumentation and recommended
         /// as the entry point for manual instrumentation.
         /// </summary>
-        public static Tracer Instance
+        public static ITracer Instance => InternalInstance;
+
+        internal static Tracer InternalInstance
```

Comment:
Copying my comment from #2059 for discussion:

I'm not sure this change is necessary 🤔  I think having a `public static Tracer Instance` is fine? I can't see the benefit of the change, and IMO it just adds confusion for us? We can just make the existing setter `internal` if that's the reason for the change (though I avoided this breaking change by adding the obsolete attribute, in case people are relying on this for some unknown reason).

For the record, my reason for suggesting the `ITracer` interface is so that people can do this setup in their DI config:

```csharp
public void ConfigureServices(IServiceCollection services)
{
    services.AddSingleton<ITracer>(() => Tracer.Instance);
}
```

And use it as 

```csharp
public class MyController
{
    private readonly ITracer tracer;
    public MyController(ITracer tracer)

    public string HelloWorld()
    {
        using var scope = tracer.StartActive("Something);
        return "HelloWorld!";
    }
}
```

That way they can mock `ITracer` in their tests.

If using `ITracer` gives us other benefits around rewriting etc, that's fair enough, but if we don't envision any, I don't know if the change is worth it

Reply:
When I wrote the code I'm not sure why I was thinking that I needed both the `Tracer : ITracer` change and the `ITracer Instance` change, but you're right that the second change is not needed. Since it's not strictly necessary and I also don't see a lot of value in keeping that, I'm in favor of removing it

Reply:
Resolving this conversation because we decided to rollback the change and just leave the `Tracer.Instance` property as-is, though it does now implement `ITracer`